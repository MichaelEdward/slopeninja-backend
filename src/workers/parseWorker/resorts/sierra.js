import cheerio from 'cheerio';

import {
  degreeOrNull,
  inchOrNull,
  numberOrNull,
  weatherStatusOrNull,
} from '../util';

const initialWeather = {
  status: null,
  weatherIcon: null,
  temprature: null,
  baseCondition: null,
  newSnow: null,
  snowDepthBase: null,
  snowDepthSummit: null,
};

const initialLifts = {
  total: null,
  open: null,
};

const initialTrails = {
  total: null,
  open: null,
};

// const initialTrails = {
//   total: null,
//   open: null,
// };

export const parseSierraSnow = async ($) => {
  const weatherIcon = $('.weather-stat-wrapper .weather-stat p').first().text().trim();
  const temprature = $('.weather-block .value').first().text().trim();
  //24 Hours
  const newSnow24Hr = $('.weather-block.weather-block-special .value').first().text().trim();
  //Base
  const snowDepthBase = $('.weather-block.weather-block-small.table-column .value').last().text().trim();
  const snowDepthSummit = $('.weather-block.weather-block-small.table-column .value').slice( 8,9 ).text().trim();
  return {
    ...initialWeather,
    weatherIcon: weatherStatusOrNull(weatherIcon),
    temprature: degreeOrNull(temprature),
    newSnow: inchOrNull(newSnow24Hr),
    snowDepthBase: inchOrNull(snowDepthBase),
    snowDepthSummit: inchOrNull(snowDepthSummit),
  };
}

export const parseSierraLifts = async ($) => {
  const openLifts = Number.parseInt($('.lift-trail-stats .lift-trail-stat .value1').first().text());
  const totalLifts = Number.parseInt($('.lift-trail-stats .lift-trail-stat .value2').first().text().replace('/', ''));
  return {
    ...initialLifts,
    total: numberOrNull(totalLifts),
    open: numberOrNull(openLifts),
  }
}

export const parseSierraTrails = async ($) => {
  const openLifts = Number.parseInt($('.lift-trail-stats .lift-trail-stat .value1').slice(1,2).text());
  const totalLifts = Number.parseInt($('.lift-trail-stats .lift-trail-stat .value2').slice(1,2).text().replace('/', ''));
  return {
    ...initialLifts,
    total: numberOrNull(totalLifts),
    open: numberOrNull(openLifts),
  }
}
