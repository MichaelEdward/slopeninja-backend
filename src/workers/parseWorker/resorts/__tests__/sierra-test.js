import fs from 'fs';
import { fetchSierra } from '../sierra';

test('fetches Sierra data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/fixtures/sierra-weather.html`);

  const resortData = await fetchSierra(htmlText);
  expect(resortData).toEqual({
    weather: {
      status: null,
      weatherIcon: null,
      temprature: 43,
      baseCondition: null,
      newSnow: 0,
      snowDepthBase: 110,
      snowDepthSummit: 211,
    },
    // lifts: {
    //   open: 0,
    //   total: 14,
    // }
  });
})

test('fetches all null for nonexisting values', async () => {
  const resortData = await fetchSierra('<html></html>');
  expect(resortData).toEqual({
    weather: {
      status: null,
      weatherIcon: null,
      temprature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    }
  });
});
