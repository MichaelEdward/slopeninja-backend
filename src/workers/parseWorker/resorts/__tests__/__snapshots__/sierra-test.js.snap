// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Sierra lift list correctly 1`] = `
Object {
  "liftList": Array [
    Object {
      "category": "Frontside Lifts",
      "name": "Easy Rider Express",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Easy Street Slider Carpet",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Easy Street Chill Carpet",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Grandview Express",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Nob Hill",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Rock Garden",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Tahoe King",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Thunder Gulch Carpet",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Tubing Hill Tow",
      "status": "closed",
    },
    Object {
      "category": "Frontside Lifts",
      "name": "Wild Mountain Carpet",
      "status": "closed",
    },
    Object {
      "category": "Backside Lifts",
      "name": "El Dorado",
      "status": "closed",
    },
    Object {
      "category": "Backside Lifts",
      "name": "Short Stuff",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "name": "Puma",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "name": "West Bowl Express",
      "status": "closed",
    },
  ],
}
`;

exports[`fetches Sierra trail list correctly 1`] = `
Object {
  "trailList": Array [
    Object {
      "category": "Rock Garden",
      "level": "beginner",
      "name": "Aspen",
      "status": "closed",
    },
    Object {
      "category": "Rock Garden",
      "level": "beginner",
      "name": "Aspen West",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Bear",
      "status": "closed",
    },
    Object {
      "category": "Easy Rider",
      "level": "beginner",
      "name": "Broadway",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Corkscrew",
      "status": "closed",
    },
    Object {
      "category": "Easy Rider",
      "level": "beginner",
      "name": "Easy Street",
      "status": "closed",
    },
    Object {
      "category": "Rock Garden",
      "level": "beginner",
      "name": "Echo",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Ego",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Lower Sleighride",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Lower Sugar n' Spice",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "beginner",
      "name": "Marten",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Royal Trail",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Spur",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "The Alley",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "beginner",
      "name": "Upper Main",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "beginner",
      "name": "Upper Snowshoe",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "beginner",
      "name": "Upper Sugar n' Spice",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "beginner",
      "name": "Wagon Trail",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Bashful",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Beaver",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Coyote",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Dogwood",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Escape",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Hemlock",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Lobo",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Lower Bashful",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "advanced",
      "name": "Lower Castle",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "advanced",
      "name": "Lower Main",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Lower Powderhorn",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Lower Snowshoe",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Marmot",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Pyramid",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Shortswing",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Smokey",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Upper Powderhorn",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "advanced",
      "name": "Upper Sleighride",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Castle",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "advanced",
      "name": "Chute",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Clipper",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Eastabout",
      "status": "closed",
    },
    Object {
      "category": "West Bowl",
      "level": "advanced",
      "name": "Horsetail",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Jackrabbit",
      "status": "closed",
    },
    Object {
      "category": "Nob Hill",
      "level": "advanced",
      "name": "Lower Dynamite",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Preacher's Passion",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Rerun",
      "status": "closed",
    },
    Object {
      "category": "Grandview",
      "level": "advanced",
      "name": "Upper Dynamite",
      "status": "closed",
    },
  ],
}
`;
